packages/yoroi-extension/chrome/extension/ergo-connector/api.js

```diff
@@ -227,7 +229,7 @@ export async function connectorSignTx(
   await RustModule.load();
   let wasmTx;
   try {
-    wasmTx = RustModule.SigmaRust.UnsignedTransaction.from_json(JSON.stringify(tx));
+    wasmTx = RustModule.SigmaRust.UnsignedTransaction.from_json(JSONBigInt.stringify(tx));
```

Comment:
If we want to keep the string values (which is what the standard specifies) we're going to want to handle that here (+ the other from_json place for boxes). Do we convert the string/number to a biginteger.js (or use the BigInt option in the lib) before calling `JSONBigInt.stringify()` on it?

Reply:
@rooooooooob I guess `number` is not really a safe option for dApps to work with tokens. As many of them have 8-9 decimals `MaxSafeNumber` often gets violated. And what JS does in that case is implicitly lowering the precision which is the worst thing in regards of working with tokens. Many dApp devs can simply forget to always check for MaxSafeNumber overflow, so I'd propose to fix the standard and step away from `number | string`.

Reply:
The best option is to replace it with `bigint | string`.

Reply:
Or, if `bigint | string` requires too heavy refactoring we can stick to `string` only.

Reply:
We will add string support to `sigma-rust` JSON decoders on monday.